sphinxmixcrypto/node.py

```diff
@@ -82,17 +62,59 @@ def get_dimensions(self):
 
 
 @attr.s(frozen=True)
-class SphinxPacket(object):
+class SphinxHeader(object):
     """
-    I am a decoded sphinx packet
+    The Sphinx header.
+
+    The Sphinx paper refers to the header fields as the greek letters: alpha, beta and gamma.
     """
     alpha = attr.ib(validator=attr.validators.instance_of(bytes))
     beta = attr.ib(validator=attr.validators.instance_of(bytes))
     gamma = attr.ib(validator=attr.validators.instance_of(bytes))
+
+
+@attr.s(frozen=True)
+class SphinxBody(object):
+    """
+    A Sphinx has the body of a lion or lioness.  The sphinx packet
+    body is repeated encrypted with the lioness wide-block cipher.
+
+    The Sphinx paper refers to this field of the packet as the greek letter delta.
+    """
     delta = attr.ib(validator=attr.validators.instance_of(bytes))
 
 
 @attr.s(frozen=True)
+class SphinxPacket(object):
+    """
+    I am a decoded sphinx packet
+    """
+    header = attr.ib(validator=attr.validators.instance_of(SphinxHeader))
+    body = attr.ib(validator=attr.validators.instance_of(SphinxBody))
+
+    def get_raw_bytes(self):
+        """
+        Get all the bytes.
+        """
+        return self.header.alpha + self.header.beta + \
+            self.header.gamma + self.body.delta
```

Comment:
(optional)

You _might_ prefer `b"".join((...))` here.  It _may_ perform better (fewer intermediates).  It also breaks across mutliple lines more nicely. :)  But the repeated `+` definitely expresses the intent very clearly, so maybe it's better as-is.